shell/app/modules/project/pages/pipelines/index.tsx

```diff
@@ -25,18 +25,33 @@ import PipelineProtocol from './components/pipeline-protocol';
 import './index.scss';
 
 const Pipeline = () => {
-  const [{ projectId }] = routeInfoStore.useStore((s) => [s.params]);
-  const [type, setType] = React.useState<{ key: string; rules?: string[] }>({ key: '', rules: [] });
+  const [{ projectId }, { typeKey: urlTypeKey }] = routeInfoStore.useStore((s) => [s.params, s.query]);
+  const [type, setType] = React.useState<{ key: string; rules?: string[] }>({} as { key: string; rules?: string[] });
   const { key: typeKey } = type;
   const [searchValue, setSearchValue] = React.useState('');
   const [list, loading] = getPipelineTypesList.useState();
   const [guidesList] = getGuidesList.useState();
   const [appID, setAppID] = React.useState<number | null>(null);
   const [expanded, setExpanded] = React.useState(false);
 
-  const getList = React.useCallback(() => {
-    getPipelineTypesList.fetch({ projectID: projectId });
-  }, [projectId]);
+  const getList = React.useCallback(async () => {
+    const res = await getPipelineTypesList.fetch({ projectID: projectId });
+    const { data } = res;
+    if (typeKey === undefined) {
```

Comment:
seems like  when typeKey!==undefined did nothing, why put this condition before request